test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildWithATargetPlatform.cs

```diff
@@ -90,5 +90,82 @@ public void It_fails_on_unsupported_os()
                 .And
                 .HaveStdOutContaining("NETSDK1139");
         }
+
+        [Fact]
+        public void It_fails_if_targetplatformversion_is_historical()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "It_fails_if_targetplatform_version_is_historical",
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework,
+            };
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+
+            string DirectoryBuildTargetsContent = $@"
+<Project>
+  <ItemGroup>
+    <SdkSupportedTargetPlatformVersion Include=""111.0"" OrGreaterHistoricalValue=""true"" />
+    <SdkSupportedTargetPlatformVersion Include=""222.0"" />
+  </ItemGroup>
+  <PropertyGroup>
+    <TargetPlatformVersion>111.0</TargetPlatformVersion>
+    <TargetPlatformIdentifier>ios</TargetPlatformIdentifier>
+    <TargetPlatformSupported>true</TargetPlatformSupported>
+  </PropertyGroup>
+</Project>
+";
+
+            File.WriteAllText(Path.Combine(testAsset.TestRoot, "Directory.Build.targets"), DirectoryBuildTargetsContent);
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute()
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining("NETSDK1140")
+                .And
+                .HaveStdOutContaining("111.0 is not a valid TargetPlatformVersion")
```

Comment:
We try to avoid hard-coding expected strings from resources in tests, as that won't work if we are trying to test under localization.  The tests should have access to the resources used in the messages, so you should be able to get the original resource string and format it to get the expected string.